mh/constants.py

```diff
@@ -7,21 +7,26 @@
 import struct
 from other.utils import pad
 from mh.time_utils import current_server_time, TICKS_PER_CYCLE, \
-    get_jhen_event_times, is_jhen_active
-from mh.quest_utils import \
-    QUEST_EVENT_JUMP_FOUR_JAGGI, QUEST_EVENT_BLOOD_SPORT, \
-    QUEST_EVENT_MERCY_MISSION, QUEST_EVENT_THE_PHANTOM_URAGAAN, \
-    QUEST_EVENT_WORLD_EATER, QUEST_EVENT_WHERE_GODS_FEAR_TO_TREAD, \
-    make_binary_event_quest
-from mh.arena_utils import \
-    GRUDGE_MATCH_ROYAL_LUDROTH, GRUDGE_MATCH_BIRD_BRUTE, \
-    GRUDGE_MATCH_TWO_FLAMES, GRUDGE_MATCH_LAND_LORDS
-
-
-def make_binary_type_time_events():
+    get_jhen_event_times, is_jhen_active, current_event_time_slot
+from mh.quest_utils import make_binary_event_quest, generate_flags
+
+
+def make_binary_type_time_events(binary_loader=None):
     return struct.pack(">III", *get_jhen_event_times())
 
 
+def make_event_slot(event):
+    return (make_binary_event_quest(event), event)
+
+
+def get_event_slot(slot_num, temporal_slot):
+    return lambda quest_loader: quest_loader[temporal_slot][slot_num] if slot_num < len(quest_loader[temporal_slot]) else b'\0' * 0x4B4
```

Comment:
Can't you create a function and return it instead? Moreover, this line is way too long.

Reply:
In this case, since the `slot_num` and `temporal_slot` need to be preserved, I actually needed to add them as optional parameters to the lambda function that default to themselves (see: https://stackoverflow.com/questions/938429/scope-of-lambda-functions-and-their-parameters). I have, however, split this line into several to make it shorter.